src/components/message-item/message-item.tsx

```diff
@@ -39,14 +39,23 @@ interface IMessageItemProps {
   isSelected?: boolean;
   animated?: boolean;
   observeIntersection: ObserveFn;
+  refToScroll?: React.RefObject<HTMLDivElement> | null;
 }
 
 const BLOCK_NAME = 'message';
 
 const linkedMessageTypes = [MessageLinkType.Forward, MessageLinkType.Reply];
 
 const MessageItem: React.FC<IMessageItemProps> = React.memo(
-  ({ messageId, selectedChatId, needToShowCreator, isSelected, observeIntersection, animated }) => {
+  ({
+    messageId,
+    selectedChatId,
+    needToShowCreator,
+    isSelected,
+    observeIntersection,
+    animated,
+    refToScroll,
```

Comment:
please pass refToScroll via forwardRef
it's an anti-pattern to use props to give ref for the wrapper HTML element